test/decl/protocol/special/coding/struct_codable_simple.swift

```diff
@@ -31,19 +31,24 @@ let _ = SimpleStruct.encode(to:)
 let _ = SimpleStruct.CodingKeys.self // expected-error {{'CodingKeys' is inaccessible due to 'private' protection level}}
 
 // rdar://problem/59655704 
-struct SR_12248_1: Codable { // expected-error {{type 'SR_12248_1' does not conform to protocol 'Encodable'}}
+struct SR_12248_1: Codable { 
+  // expected-error@-1 {{type 'SR_12248_1' does not conform to protocol 'Encodable'}}
+  // expected-error@-2 {{invalid redeclaration of synthesized enum case 'x'}} 
```

Comment:
I'm confused by this diagnostic. It looks like this is being emitted because of duplicate synthesized coding keys, but it's unlikely that most users would be able to deduce that. Instead, it just looks like the compiler is calling `SR_11248_1` an enum and `var x` an enum case.

I think the logic you adopt above of swapping "current" with "other," and just picking a location to output errors when both is implicit, is what's leading to this issue here. At minimum, if both declarations are implicit, I think it's worth suppressing the error, because this (IMO) is actively misleading.

Reply:
Yeah, it's confusing. I added it as a fallback but I can't think of any scenario where it would actually be helpful, so I think it makes sense to not emit an error if both declarations are implicit. I have done that now.